src/Package/Impl/DataInspect/VariableSearchTask.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Windows;
+using Microsoft.Common.Core;
+using Microsoft.VisualStudio.R.Package.Shell;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.R.Package.DataInspect {
+    internal sealed class VariableSearchTask : VsSearchTask {
+        private readonly IVsSearchCallback _callback;
+        private readonly TreeGrid _grid;
+
+        public VariableSearchTask(TreeGrid grid, uint dwCookie, IVsSearchQuery pSearchQuery, IVsSearchCallback pSearchCallback)
+            : base(dwCookie, pSearchQuery, pSearchCallback) {
+            _grid = grid;
+            _callback = pSearchCallback;
+        }
+
+        protected override void OnStartSearch() {
+            base.OnStartSearch();
+
+            var searchString = SearchQuery.SearchString;
+            if (_grid?.Items != null && !string.IsNullOrWhiteSpace(searchString)) {
+                foreach(var itemControl in _grid.Items) {
+                    var tn = itemControl as ObservableTreeNode;
+                    var model = tn?.Model?.Content as VariableViewModel;
+                    if(model != null) {
+                        if (model.Name.StartsWithOrdinal(searchString)) {
```

Comment:
I think we should do two passes here, first one using `StartsWith`, and then another one using `Contains` (so that variables that start with it are listed first, but you can also find stuff in the middle).